unified_planning/model/problem.py

```diff
@@ -538,53 +542,193 @@ def kind(self) -> "up.model.problem_kind.ProblemKind":
 
         IMPORTANT NOTE: this property does a lot of computation, so it should be called as
         seldom as possible."""
+        # Create the needed data structures
+        fluents_to_only_increase: Set["up.model.fnode.FNode"] = set()
+        fluents_to_only_decrease: Set["up.model.fnode.FNode"] = set()
+        static_fluents: Set["up.model.fluent.Fluent"] = self.get_static_fluents()
+
+        # sf_positive_negative is a map from static_fluent to a tuple of 2 booleans, where the first one
+        # says if the static values are only positive and the second one says if every static value is only negative.
+        sf_positive_or_negative: Dict["up.model.fluent.Fluent", Tuple[bool, bool]] = {}
+        for f, v in self.initial_values.items():
```

Comment:
Here you should iterate only trough `explicit_initial_values` and `fluents_defaults`, avoiding the full fluents grounding.

Reply:
Yes, but we would miss the case where the `fluents_default` don't fit the SNP kind but every fluent is actually initialized explicitly to a value that is OK. Right?

Reply:
But I think you can do that check without grounding the fluents, right?